src/darwin/frida-helper-process.vala

```diff
@@ -267,9 +264,10 @@ namespace Frida {
 				var handshake_port = new HandshakePort.local (service_name);
 
 				string[] argv = { get_resource_store ().helper.path, service_name };
-				pending_process = new Subprocess.newv (argv, SubprocessFlags.INHERIT_FDS);
 
-				var peer_pid = (uint) uint64.parse (pending_process.get_identifier ());
+				GLib.SpawnFlags flags = GLib.SpawnFlags.LEAVE_DESCRIPTORS_OPEN | /*GLib.SpawnFlags.CLOEXEC_PIPES*/ 256;
```

Comment:
I don't know why, but apparently `GLib.SpawnFlags.CLOEXEC_PIPES` doesn't seem to be exposed (even if it's documented) and i get an error like:
```
../../../frida-core/src/darwin/frida-helper-process.vala:266.70-266.98: error: The name `CLOEXEC_PIPES' does not exist in the context of `GLib.SpawnFlags' (glib-2.0)
				GLib.SpawnFlags flags = GLib.SpawnFlags.LEAVE_DESCRIPTORS_OPEN | GLib.SpawnFlags.CLOEXEC_PIPES;
				                                                                 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
```

Reply:
i'm not even sure we really need it, i kept it for compatibility with what SubProcess was already doing here.